node_iter.go

```diff
@@ -10,11 +12,10 @@ type nodeIter struct {
 	li  Iterator
 }
 
-func newNodeIter(base []*node) *nodeIter {
-	return &nodeIter{
-		stk: [][]*node{base},
-		li:  exhaustedIterator{},
-	}
+func newNodeIter(base []*node, count int) *nodeIter {
+	depth := (bits.Len64(uint64(count)) + 5) / 2 // 1.5 (log₈(n) + 1)
+	stk := append(make([][]*node, 0, depth), base)
+	return &nodeIter{stk: stk, li: exhaustedIterator{}}
```

Comment:
trying to understand this part, why the `count` argument? is it also for parallelisation?

Reply:
It's to compute a max-depth estimate in order to preallocate the stack.